### PR TITLE
Font: Fix to have correct fonts

### LINKED ISSUE #1950: Change `<h3>`, `<h5>`, and `<h6>` values
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

In ["Applying the type styles" DSM section](https://system.design.orange.com/0c1af118d/p/38f221-typography/t/08c35b), we can see that the values for `<h3>`, `<h5>` and `<h6>` are not the same as the one proposed in Boosted.

Currently, we have:

  * `<h3>`: Desktop(24px) / Tablet(18px) / Mobile(16px)
  * `<h5>`: Desktop(18px) / Tablet(16px) / Mobile(16px)
  * `<h6>`: Desktop(16px) / Tablet(16px) / Mobile(16px)

But based on the DSM, they should be:

  * `<h3>`: Desktop(24px) = Subtitle 2 / **Tablet(24px)** = Subtitle 1 / **Mobile(18px)** = Subtitle
  * `<h5>`: Desktop(18px) = Body 3 / **Tablet(18px)** = Body 2 / Mobile(16px) = Body 2
  * `<h6>`: **Desktop(18px)** = Body 4 / **Tablet(18px)** = Body 2 / Mobile(16px) = Body 2

⚠️ `<h2>` and `<h3>` will have the same size on tablet and mobile. It's not fundamentally wrong because designers and/or developers can play with the colors, font weights, etc. to differentiate the heading levels. It might be worth adding this information to our documentation!

### What operating system(s) are you seeing the problem on?

Windows, macOS, Android, iOS, Linux

### What browser(s) are you seeing the problem on?

Chrome, Safari, Firefox, Microsoft Edge, Opera

### What version of Boosted are you using?

v5.2

### PR DESCRIPTION
**Pull request by louismaximepiton**
Note: Please transform - [ ] into - (NA) in the description when things are not applicable 

### Related issues 

Fixes #1950. 

### Description 

Change breakpoints of application to better fit design rules. 
Change font sizes depending on the breakpoint to better suit design. 
Update the doc to better convey information.

In ["Applying the type styles" DSM section](https://system.design.orange.com/0c1af118d/p/38f221-typography/t/08c35b), we can see that the values for `<h3>`, `<h5>` and `<h6>` are not the same as the one proposed in Boosted atm.

Currently, we have:

    `<h3>`: Desktop(24px) / Tablet(18px) / Mobile(16px)
    `<h5>`: Desktop(18px) / Tablet(16px) / Mobile(16px)
    `<h6>`: Desktop(16px) / Tablet(16px) / Mobile(16px)

But based on the DSM, they should be:

    `<h3>`: Desktop(24px) = Subtitle 2 / Tablet(24px) = Subtitle 1 / Mobile(18px) = Subtitle
    `<h5>`: Desktop(18px) = Body 3 / Tablet(18px) = Body 2 / Mobile(16px) = Body 2
    `<h6>`: Desktop(18px) = Body 4 / Tablet(18px) = Body 2 / Mobile(16px) = Body 2

⚠️ `<h2>` and `<h3>` will have the same size on tablet and mobile. `<h5>` and `<h6>` will have the exact same size everytime.

**Dear a11y experts and designers**, please check that the titles of the live preview below are fine (please resize your screen to fully check the styles). Make sure to also check the breakpoints that have been changed too.

Don't forget to check https://deploy-preview-2053--boosted.netlify.app/docs/examples/stickers and https://deploy-preview-2053--boosted.netlify.app/docs/5.3/components/sticker/

### Motivation & Context 

Wrong display atm. 

### Types of change 

- Bug fix (non-breaking which fixes an issue) 

### Live previews 

- https://deploy-preview-2053--boosted.netlify.app/docs/content/typography#headings to compare with https://main--boosted.netlify.app/docs/5.3/content/typography#headings

#### Contribution 

- [x] I have read the contributing guidelines 

#### Accessibility 

- [x] My change follows accessibility good practices; I have at least run axe 

#### Design 
- [x] My change respects the design guidelines defined in Orange Design System 
- [x] My change is compatible with responsive display 

#### Development 

- [x] My change follows the developer guide 
- (NA) I have added JavaScript unit tests to cover my changes 
- (NA) I have added SCSS unit tests to cover my changes 
 
#### Documentation 

- [x] My change introduces changes to the documentation and/or I have updated the documentation accordingly 

### Checklist (for Core Team only) 

- [x] My change introduces changes to the migration guide 
- (NA) My new component is added in Storybook 
- (NA) My new component is compatible with RTL 
- [ ] Manually run BrowserStack tests 
- [ ] Manually test browser compatibility with BrowserStack (Chrome >= 60, Firefox >= 60 (+ ESR), Edge, Safari >= 12, iOS Safari, Chrome & Firefox on Android) 
- [ ] Code review 
- [ ] Design review 
- [ ] A11y review 

#### After the merge 

- [ ] Manually launch Percy tests

### CODE REVIEW

**Review by MewenLeHo — APPROVED** [review]
Font size values are ok.

I am just not really confortable with the fact that some titles with differents heading levels can have the same size. It may be visually confusing for some people even if the semantic behind it is correct.

Therefore I would recommend an a11y review for this PR.

**Comment by Aniort** [comment]
a11y checked, OK

**Review by julien-deramond — CHANGES_REQUESTED** [review]
FYI, this PR modifies the rendering of the stickers which use `<h3>`s. I don't remember the specifications but is it OK?

On a more general non-regression approach, have you checked in all the documentation, components, and examples, that the modification of `<h3>`, `<h4>`, and `<h5>` doesn't cause any issues?

**Comment by louismaximepiton** [comment]
Good catchs! 
Corrections are done via last commit.

For instance, impacts I could get on the doc: 
- Stickers that are using `class="h3"`
- List-group headings that are out of DSM
- Carousel slide labels that are out of DSM

The remaining seems to be already overrided or directly linked to the documentation (footer, cards, offcanvas, modals, .

This PR could lead to resolve some of the issues we had such as #786, `.modal-title`, `offcanvas-title` and maybe more?

**Comment by julien-deramond** [comment]
For stickers, we must check with the designers.
For stuff not described in the DSM, default rendering for `<h*>`s should be OK.

> This PR could lead to resolve some of the issues we had such as https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/786, .modal-title, offcanvas-title and maybe more?

If there are no regressions in the PR, let's not modify them yet because, with the new tokens, we will need to modify them again right after. With your modification, does it have an impact or does it stay with the same rendering?

**Comment by louismaximepiton** [comment]
There is no regression because we override the style in those classes, but I wanted to mention that we could think of ease our styles due to these changes. However, you're right and we should wait for tokens.

File: site/content/docs/5.3/migration.md [hunk 1]
```diff
@@ -138,6 +138,9 @@ Learn more by reading the new [color modes documentation]({{< docsref "/customiz
 
 - **Typography**
   - We no longer set a color for `$headings-color-dark` or `--bs-heading-color` for dark mode. To avoid several problems of headings within components appearing the wrong color, we've set the Sass variable to `null` and added a `null` check like we use on the default light mode.
+  - <span class="badge bg-warning">Breaking</span> Some headings font size values have been changed to be compliant with the design guidelines. Please check it has no impact on your websites.
+    - `<h3>` values are now `24px` for desktop and tablet breakpoints and `18px` for mobile breakpoint.
```
Comment (julien-deramond) [inline site/content/docs/5.3/migration.md:142]: For these 2 sentences, could you double-check, we'd like to mention only what has changed; in terms of values and not in terms of documentation.

If I refer to the documentation only of the current main branch (which is probably wrong), `<h3>` being `24px` for desktop is not new, and `<h5>` being `18px` for desktop neither, for example.

Comment (louismaximepiton) [reply site/content/docs/5.3/migration.md:142]: After spec review, we'll need to fix font size.

**Comment by louismaximepiton** [comment]
Reverted the change of breakpoints as it could be potentially breaking.

**Comment by julien-deramond** [comment]
@MewenLeHo A lot of modifications happened since your first review regarding the stickers. Could you please take a look at what's been done to see if you agree with this new way of handling the stickers? No need to check in detail, I'll do it. Just talking about the principle to rely on classes not embedded in the framework.

**Comment by MewenLeHo** [comment]
> @MewenLeHo A lot of modifications happened since your first review regarding the stickers. Could you please take a look at what's been done to see if you agree with this new way of handling the stickers? No need to check in detail, I'll do it. Just talking about the principle to rely on classes not embedded in the framework.

I am honestly not in favor of this solution.
Boosted's goal is to give users 'reusable prebuild components' so for me all component's related styles must be embbed in the framework.

**Comment by julien-deramond** [comment]
> Boosted's goal is to give users 'reusable prebuild components' so for me all component's related styles must be embbed in the framework.

The issue here is that it works "well" for these examples, but the sizes won't ever be the same depending on the use cases: content, internationalization, etc. So we can't predict the font size values and all the combinations.
Do you have another approach to suggest here?

**Comment by Franco-Riccitelli** [comment]
Font sizes
Values for `<h3>`, `<h5>` and `<h6>`. These values will also need to be updated in the DSM.
```
`<h3>`: Desktop(24px) / Tablet(24px) / Mobile(18px)
`<h5>`: Desktop(18px) / Tablet(18px) / Mobile(16px)
`<h6>`: Desktop(16px) / Tablet(16px) / Mobile(16px)
```

**Comment by Franco-Riccitelli** [comment]
The implementation of the Stickers all looks good. No issues.

**Comment by julien-deramond** [comment]
@louismaximepiton https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2053/commits/2fb489db93b864269eafd7e9f76f23c993519f60 is a proposal to use `scss-docs` to handle `.sticker-fs-*` classes in Sticker documentation. It'll avoid maintenance if those values evolve in the future since we'll have to update them only in the CSS used for the docs. Can be reverted if you think this is not OK.

File: site/assets/scss/_component-examples.scss [hunk 2]
```diff
@@ -400,6 +400,39 @@
...
+// scss-docs-end sticker-fs-l-3xl
+
+// scss-docs-start sticker-fs-xl
+.sticker-fs-xl {
+  font-size: 1.875rem;
+  line-height: calc(32 / 30); // stylelint-disable-line function-disallowed-list
+  letter-spacing: -.05rem;
+}
+// scss-docs-end sticker-fs-xl
+// scss-docs-end custom-fdsf
```
Comment (julien-deramond) [inline site/assets/scss/_component-examples.scss:434]: Oups

**Review by julien-deramond — APPROVED** [review]
LGTM 🚀

**Pull request merged**